politeiawww/convert.go

```diff
@@ -814,3 +815,178 @@ func convertInvoiceFromCache(r cache.Record) cms.InvoiceRecord {
+		switch v.ID {
+		case mdStreamDCCGeneral:
+			// General invoice metadata
+			m, err := decodeBackendDCCMetadata([]byte(v.Payload))
+			if err != nil {
+				log.Errorf("convertDCCFromCache: decode md stream: "+
+					"token:%v error:%v payload:%v",
+					r.CensorshipRecord.Token, err, v)
+			}
+			md = *m
+
+		case mdStreamDCCStatusChanges:
+			// Invoice status changes
+			m, err := decodeBackendDCCStatusChanges([]byte(v.Payload))
+			if err != nil {
+				log.Errorf("convertDCCFromCache: decode md stream: "+
+					"token:%v error:%v payload:%v",
+					r.CensorshipRecord.Token, err, v)
+			}
+
+			// We don't need all of the status changes.
+			// Just the most recent one.
+			for _, s := range m {
+				c = s
+			}
+		}
+	}
+
+	// Convert files
+	var di cms.DCCInput
+	fs := make([]www.File, 0, len(r.Files))
+	for _, v := range r.Files {
+		f := www.File{
+			Name:    v.Name,
+			MIME:    v.MIME,
+			Digest:  v.Digest,
+			Payload: v.Payload,
+		}
+		fs = append(fs, f)
+
+		// Parse invoice json
+		if f.Name == dccFile {
+			b, err := base64.StdEncoding.DecodeString(v.Payload)
+			if err != nil {
+				log.Errorf("convertDCCFromCache: decode invoice: "+
+					"token:%v error:%v payload:%v",
+					r.CensorshipRecord.Token, err, f.Payload)
+				continue
+			}
+
+			err = json.Unmarshal(b, &di)
+			if err != nil {
+				log.Errorf("convertDCCFromCache: unmarshal InvoiceInput: "+
+					"token:%v error:%v payload:%v",
+					r.CensorshipRecord.Token, err, f.Payload)
+				continue
+			}
+		}
+	}
+
+	// UserID and Username are left intentionally blank.
+	// These fields not part of a cache record.
+	return cms.DCCRecord{
+		Status:             c.NewStatus,
+		StatusChangeReason: c.Reason,
+		Timestamp:          r.Timestamp,
+		SponsorUserID:      "",
+		SponsorUsername:    "",
+		PublicKey:          md.PublicKey,
+		Signature:          md.Signature,
+		Files:              fs,
+		Version:            r.Version,
+		CensorshipRecord: www.CensorshipRecord{
+			Token:     r.CensorshipRecord.Token,
+			Merkle:    r.CensorshipRecord.Merkle,
+			Signature: r.CensorshipRecord.Signature,
+		},
+		DCC: di,
+	}
+}
+
+func convertRecordToDatabaseDCC(p pd.Record) (*cmsdatabase.DCC, error) {
+	dbDCC := cmsdatabase.DCC{
+		Files:           convertRecordFilesToWWW(p.Files),
+		Token:           p.CensorshipRecord.Token,
+		ServerSignature: p.CensorshipRecord.Signature,
+		Version:         p.Version,
+	}
+
+	// Decode invoice file
+	for _, v := range p.Files {
+		if v.Name == dccFile {
+			b, err := base64.StdEncoding.DecodeString(v.Payload)
+			if err != nil {
+				return nil, err
+			}
+
+			var dcc cms.DCCInput
+			err = json.Unmarshal(b, &dcc)
+			if err != nil {
+				return nil, www.UserError{
+					ErrorCode: www.ErrorStatusInvalidInput,
+				}
+			}
+			dbDCC.Type = dcc.Type
+			dbDCC.NomineeUserID = dcc.NomineeUserID
+			dbDCC.SponsorStatement = dcc.SponsorStatement
+			dbDCC.Domain = dcc.Domain
+		}
+	}
+
+	for _, m := range p.Metadata {
```

Comment:
I think you need to add the new mdstreams to the switch statement.  I'm getting these errors when submitting a DCC:


```
2019-08-20 12:26:45.848 [ERR] PWWW: initializeInventory: invalid metadata stream ID 5 token ed7d6eceb1ff437aa449c6abd9a565d60162f8e8649c88bd27994efde5dd930c
2019-08-20 12:26:45.848 [ERR] PWWW: initializeInventory: invalid metadata stream ID 6 token ed7d6eceb1ff437aa449c6abd9a565d60162f8e8649c88bd27994efde5dd930c
```